modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -763,6 +774,25 @@ void ONNXImporter::populateNet(Net dstNet)
             layerParams.set("end_axis", axis);
             layerParams.type = "Flatten";
         }
+        else if (layer_type == "Flatten")
+        {
+            CV_Assert_N(node_proto.input_size() == 1);
+            if (constBlobs.find(node_proto.input(0)) != constBlobs.end())
+            {
+                int axis = layerParams.get<int>("axis", 1);
+                Mat input = getBlob(node_proto, constBlobs, 0);
+                axis = clamp(axis, input.dims);
+
+                std::vector<int> out_size(&input.size[0], &input.size[0] + axis);
+                int last_dim = input.total(axis);
+                out_size.push_back(last_dim);
+
+                Mat output = input.reshape(1, out_size);
+                constBlobs.insert(std::make_pair(layerParams.name, output));
+                continue;
+            }
+
+        }
```

Comment:
Is there Flatten with variable inputs?

Reply:
Always input.size() = 1

Reply:
But the single output may be as variable as constant. For constant we can calculate the output to constBlobs, for variable input we need to create "Flatten" layer.

Reply:
For variable input we use "Flatten" layer created with default params.